packages/drizzle/src/utilities/buildCreateMigration.ts

```diff
@@ -0,0 +1,134 @@
+import type { DrizzleSnapshotJSON } from 'drizzle-kit/api'
+import type { CreateMigration } from 'payload'
+
+import fs from 'fs'
+import path from 'path'
+import { getPredefinedMigration, writeMigrationIndex } from 'payload'
+import prompts from 'prompts'
+
+import type { DrizzleAdapter } from '../types.js'
+
+import { getMigrationTemplate } from './getMigrationTemplate.js'
+
+export const buildCreateMigration = ({
+  executeMethod,
+  filename,
+  sanitizeStatements,
+}: {
+  executeMethod: string
+  filename: string
+  sanitizeStatements: (args: { sqlExecute: string; statements: string[] }) => string
+}): CreateMigration => {
+  const dirname = path.dirname(filename)
+  return async function createMigration(
+    this: DrizzleAdapter,
+    { file, forceAcceptWarning, migrationName, payload, skipEmpty },
+  ) {
+    const dir = payload.db.migrationDir
+    if (!fs.existsSync(dir)) {
+      fs.mkdirSync(dir)
+    }
+
+    const { generateDrizzleJson, generateMigration, upSnapshot } = this.requireDrizzleKit()
+
+    const drizzleJsonAfter = await generateDrizzleJson(this.schema)
+
+    const [yyymmdd, hhmmss] = new Date().toISOString().split('T')
+    const formattedDate = yyymmdd.replace(/\D/g, '')
+    const formattedTime = hhmmss.split('.')[0].replace(/\D/g, '')
+    let imports: string = ''
+    let downSQL: string
+    let upSQL: string
+    ;({ downSQL, imports, upSQL } = await getPredefinedMigration({
```

Comment:
Can you make a canary and verify in a project that it still works with the same command?